homeassistant/components/hive/alarm_control_panel.py

```diff
@@ -0,0 +1,103 @@
+"""Support for the Hive alarm."""
+from datetime import timedelta
+
+from homeassistant.components.alarm_control_panel import AlarmControlPanelEntity
+from homeassistant.components.alarm_control_panel.const import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_NIGHT,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+
+from . import HiveEntity
+from .const import DOMAIN
+
+ICON = "mdi:security"
+PARALLEL_UPDATES = 0
+SCAN_INTERVAL = timedelta(seconds=15)
+HIVETOHA = {
+    "home": STATE_ALARM_DISARMED,
+    "asleep": STATE_ALARM_ARMED_NIGHT,
+    "away": STATE_ALARM_ARMED_AWAY,
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Hive thermostat based on a config entry."""
+
+    hive = hass.data[DOMAIN][entry.entry_id]
+    devices = hive.session.deviceList.get("alarm_control_panel")
+    entities = []
+    if devices:
+        for dev in devices:
+            entities.append(HiveAlarmControlPanelEntity(hive, dev))
+    async_add_entities(entities, True)
```

Comment:
The scope of variables could be reduced here.

Or maybe the entire thing can be shortened. I think something like this can work?
```suggestion
    if devices:
        async_add_entities([HiveAlarmControlPanelEntity(hive, dev) for dev in devices], True)
```